cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -659,6 +660,33 @@ public void getApplicationDetails(FullHttpRequest request, HttpResponder respond
     }
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(result));
   }
+  /**
+   * Returns the source control metadata and sync status of all applications
+   * filter query format - "name=<name-filter>ANDsyncStatus=<SYNCED/UNSYNCED>"
+   */
+  @GET
+  @Path("/appscmstatus")
+  public void getAllAppsSCMStatus(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace,
+      @QueryParam("pageToken") String pageToken,
+      @QueryParam("pageSize") Integer pageSize,
+      @QueryParam("orderBy") SortOrder orderBy,
+      @QueryParam("orderByOption") SortOrderOptions orderByOption,
+      @QueryParam("filter") String filter
+     ) throws Exception {
+    // TODO(adrikagupta): Implement the API handler
+  }
+
+  /**
+   * Returns the source control metadata and sync status of a specific application
+   */
+  @GET
+  @Path("/apps/{app-id}/scmstatus")
+  public void getAppSCMStatus(HttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") final String namespaceId,
+      @PathParam("app-id") final String appName) throws Exception {
+    // TODO(adrikagupta): Implement the API handler
```

Comment:
Please add a `JIRA ID` for all the TODOs, instead of usernames.

Reply:
Sure, done